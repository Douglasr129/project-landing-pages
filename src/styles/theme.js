export const theme = {
	colors: {
		mainBg: 'red',
		secundaryBg: 'blue',
	},
	fonts: {},
	spacing: {},
};
