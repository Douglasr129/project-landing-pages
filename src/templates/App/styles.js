import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
	${(props) => css`
			background: ${props.background};
	`}
	font-size: 1rem;
	font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif
`;
