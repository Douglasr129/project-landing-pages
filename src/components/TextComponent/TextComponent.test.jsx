import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { TextComponent } from '.';
import { theme } from '../../styles/theme';

describe('<TextComponent />', () => {
	it('should render a text', () => {
		renderTheme(<TextComponent>Children</TextComponent>);
		expect(screen.getByText('Children')).toBeInTheDocument();
	});
	it('should render a dark color', () => {
		renderTheme(<TextComponent>Children</TextComponent>);
		const textComponent = screen.getByText('Children');
		expect(textComponent).toHaveStyle({
			color: theme.colors.primaryColor,
			'font-size': theme.font.sizes.medium,
		});
	});
	it('should render with whote color', () => {
		renderTheme(<TextComponent colorDark={false}>Children</TextComponent>);
		const textComponent = screen.getByText('Children');
		expect(textComponent).toHaveStyle({ color: theme.colors.white });
	});
	it('should render with whote color', () => {
		const { container } = renderTheme(<TextComponent>Children</TextComponent>);
		expect(container.firstChild).toMatchSnapshot();
	});
});
