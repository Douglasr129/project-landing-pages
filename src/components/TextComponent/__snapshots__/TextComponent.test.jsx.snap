// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextComponent /> should render with whote color 1`] = `
.c0 {
  color: #0a1128;
  font-size: 2.4rem;
}

<div
  class="c0"
>
  Children
</div>
`;
